### ISSUE TITLE
Resume from hibernate fails in LVM on LUKS setup

### ISSUE THREAD

**Issue by bernerdschaefer** [NixOS/nixpkgs#8803]
My LVM setup is described here: https://robots.thoughtbot.com/install-linux-on-a-macbook-air#configure-disk-encryption

  - luks-encrypted disk
  - lvm managed root and swap

After powering back on after hibernate, no system state is restored. Nothing relevant seems to be logged.

This appears to be a timing issue, in that in [stage-1-init.sh] the resume device is checked immediately after updating udev and lvm, while later steps sleep and retry to await the device.

Confirmed by:
  - booting into debug1devices mode. By the time I can type the resume command being executed, it succeeds.

On that hunch, I added this to my configuration.nix:

```
boot.initrd.postDeviceCommands = "sleep 5";
```

And I now can resume without problems. I'm not familiar enough with udev or lvm to provide an alternative, but it would be great to have the stage-1-init script support this setup without hacks like the above.

  [stage-1-init.sh]: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/stage-1-init.sh

**Comment by Pleune** [comment]
I can confirm that this fix works. However, I'm running a very minimal/default configuration. No disk encrypting, just a standard install.

**Comment by j1r1k** [comment]
I was facing issues with hibernation using similar configuration. I managed to solve it by explicitly declaring swap device to be used as a resume device
```
boot.resumeDevice = "/dev/disk/by-label/swap";
```

**Comment by Pleune** [comment]
@j1r1k I believe I had tried this at the time, and found that it didn't fix my problem or was it necessary. I found that my issue was hat my laptop (hp stream) has an emmc hard drive, and similar to the LVM setup described above, stage-1-init.sh was not waiting for the driver to come up. 

I could find no other fix than adding a delay, and this is still in my config. I am not sure if the issue has been resolved.
